src/test/java/at/ac/tuwien/kr/alpha/solver/SolverTests.java

```diff
@@ -27,174 +27,122 @@
-			.predicate("q").instance("1").instance("2")
-			.predicate("p").instance("1").instance("2")
-			.build();
-
-		assertEquals(expected, answerSets.get(0));
+			"q(1), q(2), p(1), p(2)"
+		);
 	}
 
 	@Test
 	public void testProgramZeroArityPredicates() throws Exception {
-		String testProgram = "a. p(X) :- b, r(X).";
-		Program parsedProgram = parseVisit(testProgram);
-		Grounder grounder = new NaiveGrounder(parsedProgram);
-		Solver solver = getInstance(grounder);
-
-		List<AnswerSet> answerSets = solver.collectList();
+		assertAnswerSet(
+			"a. p(X) :- b, r(X).",
 
-		assertEquals(1, answerSets.size());
-
-		AnswerSet expected = new BasicAnswerSet.Builder()
-			.predicate("a")
-			.build();
-
-		assertEquals(expected, answerSets.get(0));
+		"a"
+		);
 	}
 
 	@Test
 	public void testGuessingGroundProgram() throws Exception {
-		Solver solver = getInstance(new NaiveGrounder(parseVisit("a :- not b. b :- not a.")));
-
-		Set<AnswerSet> expected = new HashSet<>(Arrays.asList(
-			new BasicAnswerSet.Builder().predicate("a").build(),
-			new BasicAnswerSet.Builder().predicate("b").build()
-		));
+		assertAnswerSets(
+			"a :- not b. b :- not a.",
 
-		assertEquals(expected, solver.collectSet());
+			"a",
+			"b"
+		);
 	}
 
 	@Test
 	public void testGuessingProgramNonGround() throws Exception {
-		String testProgram = "dom(1). dom(2). dom(3)." +
+		assertAnswerSetsWithBase(
+			"dom(1). dom(2). dom(3)." +
 			"p(X) :- dom(X), not q(X)." +
-			"q(X) :- dom(X), not p(X).";
-		Program parsedProgram = parseVisit(testProgram);
-		NaiveGrounder grounder = new NaiveGrounder(parsedProgram);
-		Solver solver = getInstance(grounder);
-
-		final BasicAnswerSet.Builder base = new BasicAnswerSet.Builder()
-			.predicate("dom").instance("1").instance("2").instance("3");
-
-		Set<AnswerSet> expected = new HashSet<>(Arrays.asList(
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("1").instance("2")
-				.predicate("p").instance("3")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("1")
-				.predicate("p").instance("2").instance("3")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("2")
-				.predicate("p").instance("1").instance("3")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("p").instance("1").instance("2").instance("3")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("1").instance("2").instance("3")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("1").instance("3")
-				.predicate("p").instance("2")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("2").instance("3")
-				.predicate("p").instance("1")
-				.build(),
-			new BasicAnswerSet.Builder(base)
-				.predicate("q").instance("3")
-				.predicate("p").instance("1").instance("2")
-				.build()
-		));
-
-		assertEquals(expected, solver.collectSet());
+			"q(X) :- dom(X), not p(X).",
```

Comment:
This is a really neat syntax now. Impressive!

Reply:
Yeah. It's probably a bit slower, because the answer sets need to be parsed, but it is _much_ cleaner. I also thought about having our tests file based, for example as follows:

```
c :- not a, not b.

%< SNIP %< SNAP

{ a }
{ b }
{ c }
```

JUnit could then just try all of them.

Reply:
Yes, one could put some tests in files. But I think it is more convenient to have them as source, because if a test fails, I can simply start debugging it with a right click. If it is file based, I assume there is only one entry point for all tests and this entry must be configured correctly to start the desired test. And currently I see no advantage of having each test in its own file(s).

Reply:
Alright.